rgw/v2/tests/s3cmd/reusable.py

```diff
@@ -33,16 +33,17 @@ def create_bucket(bucket_name):
     assert expected_response in mb_response, error_message
 
 
-def upload_file(bucket_name):
+def upload_file(bucket_name, file_name=None):
     """
     Uploads file to the bucket
     Args:
         bucket_name(str): Name of the bucket
     Returns: Name of the uploaded file
     """
-    file_name = "test_s3cmd.txt"
-    with open(file_name, "w") as f:
-        f.write("Test file")
+    if file_name is None:
```

Comment:
We could avoid this check by doing

```
def upload_file(bucket_name, file_name="test_s3cmd.txt"):
  with open(file_name, "w") as f: 
    f.write("test file")
```

Reply:
I think you are trying to upload a file to the cloud. can you use this function to generate [IO]( https://github.com/red-hat-storage/ceph-qe-scripts/blob/master/rgw/v2/lib/manage_data.py#L13) and also check [example](https://github.com/red-hat-storage/ceph-qe-scripts/blob/master/rgw/v2/tests/s3_swift/test_Mbuckets_with_Nobjects.py#L167 ) for upload implementation, you could wrap something similar in s3cmd test.

Reply:
@psathyan , we are passing file name if we have file already in place. From testcase, I'm passing 2GB file which is already available in local directory. So, it shouldn't do write a content to that file.

Reply:
Thanks. Now, it is clear on why you are having the check.

Reply:
@rakeshgm , As we are testing the functionality with s3cmd, I have used this way of uploading object using s3cmd itself. I believe it will give coverage for uploading large object using s3cmd too which will help in finding issue in large object upload using s3cmd. Please let me know your input on this.